File: src/main/java/com/wgzhao/addax/admin/controller/IndexController.java
package com.wgzhao.addax.admin.controller;

import com.wgzhao.addax.admin.service.SourceService;
import com.wgzhao.addax.admin.service.StatService;
import com.wgzhao.addax.admin.service.TableService;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

/**
 * 监控主页面，主要是汇总信息和图表展示
 */
@RestController
@RequestMapping("/dashboard")
@AllArgsConstructor
public class IndexController
{

    private final StatService statService;
    private final SourceService sourceService;
    private final TableService tableService;

    // 各数据源采集完成率，用于图表展示
    @RequestMapping("/accomplish-ratio")
    public ResponseEntity<List<Map<String, Object>>> accomplishRatio()
    {
        return ResponseEntity.ok(statService.statLastAccompRatio());
    }

    //  最近5天采集耗时对比
    @RequestMapping("/last-5d-collect-time")
    public ResponseEntity<List<Map<String, Object>>> last5DaysEtlTime()
    {
        return ResponseEntity.ok(statService.statLast5DaysTimeBySource());
    }

    // 最近交易日采集的数据量， 以 GB 为单位
    @RequestMapping("/last-collect-data")
    public ResponseEntity<Double> lastEtlData()
    {
        return ResponseEntity.ok(statService.statTotalData());
    }

    // 获取最近12个月的采集累计数据量，单位为 GiB
    @RequestMapping("/last-12m-collect-data")
    public ResponseEntity<?> last12MonthsData()
    {
        // Adjusted method signature for consistency
        return ResponseEntity.ok(statService.statLast12MonthsData());
    }

    @RequestMapping("/collect-table-count")
    public ResponseEntity<Integer> tableCount()
    {
        return ResponseEntity.ok(tableService.getValidTableCount());
    }

    @GetMapping("/collect-source-count")
    public ResponseEntity<Integer> sourceCount()
    {
        return ResponseEntity.ok(sourceService.getValidSources());
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/controller/IndexController.kt
package com.wgzhao.addax.admin.controller

import com.wgzhao.addax.admin.service.SourceService
import com.wgzhao.addax.admin.service.StatService
import com.wgzhao.addax.admin.service.TableService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * 监控主页面控制器，主要用于汇总信息和图表展示
 */
@RestController
@RequestMapping("/dashboard")
class IndexController(
    private val statService: StatService,
    private val sourceService: SourceService,
    private val tableService: TableService
) {
    /**
     * 获取各数据源采集完成率，用于图表展示
     * @return 完成率列表
     */
    @GetMapping("/accomplish-ratio")
    fun accomplishRatio(): ResponseEntity<List<Map<String, Any>>> =
        ResponseEntity.ok(statService.statLastAccompRatio())

    /**
     * 获取最近5天采集耗时对比
     * @return 耗时数据列表
     */
    @GetMapping("/last-5d-collect-time")
    fun last5DaysEtlTime(): ResponseEntity<List<Map<String, Any>>> =
        ResponseEntity.ok(statService.statLast5DaysTimeBySource())

    /**
     * 获取最近交易日采集的数据量（单位GB）
     * @return 数据量
     */
    @GetMapping("/last-collect-data")
    fun lastEtlData(): ResponseEntity<Double> =
        ResponseEntity.ok(statService.statTotalData())

    /**
     * 获取最近12个月采集累计数据量（单位GiB）
     * @return 月度数据量列表
     */
    @GetMapping("/last-12m-collect-data")
    fun last12MonthsData(): ResponseEntity<List<Map<String, Any>>> =
        ResponseEntity.ok(statService.statLast12MonthsData())

    /**
     * 获取采集表数量
     * @return 表数量
     */
    @GetMapping("/collect-table-count")
    fun tableCount(): ResponseEntity<Int> =
        ResponseEntity.ok(tableService.validTableCount)

    /**
     * 获取数据源数量
     * @return 数据源数量
     */
    @GetMapping("/collect-source-count")
    fun sourceCount(): ResponseEntity<Int> =
        ResponseEntity.ok(sourceService.validSources)
}
